### DISCUSSION TITLE
Allow some habits to receive a checkmark automatically at the beginning of the day

### DISCUSSION THREAD

**Discussion by ghost** [iSoron/uhabits#125] [Feature Requests]
just a thought. maybe some habits would be better suited if there was an option to have the checkmarks ticked automatically instead of having to do it manually every day? for things were you know you will be doing them most days of the week and only be missing every once in a while. it would be easier in those cases just to untick the habit whenever you miss it.

one example im thinking of is trying to give up chocolate. sometimes i might go weeks without having some, but it means that i still have to go to the app every day and hit the button. its not really a problem with just one, but when there is a whole screen of habits that have to be ticked off each day it can start to seem like a chore.

**Comment by ghost** [comment]
actually on second thoughts, you can delete this if you want. i just noticed you might be adding tasker support at some stage so being able to automatically tick off certain habits using tasker would work just as well

**Comment by iSoron** [comment]
I think this is an interesting feature, even if we have Tasker integration.

**Comment by brunosabenca** [comment]
Even though a habit might already be established, I still find value in keeping track of its streak. 

Particularly for habits that consist on abstaining from something, not wanting to break a long streak of successes can help prevent a relapse and while you might not need (or want) to mark a habit manually everyday it can still be beneficial to look at the entry every once in a while, maybe in a moment of weakness, and see how far you have come.

**Comment by YPetremann** [comment]
Ticking on an abstraining habit is quite painfull, but ticking when you do the "forbiden task" is easier, just need to be serious.

**Comment by devurandom** [comment]
This might be one way to solve #96.

**Comment by logicalup** [comment]
I think there's a way for everyone to be happy: allow user to set what to default each habit to one of three options: 1. Unchecked/unmarked, 2. Fail mark ❌or 3. Success mark ✅

**Reply by herrbrixter** [reply]
This is great for Yes or No habits. The 3rd option, default to Checked/Success is missing.

**Comment by herrbrixter** [comment]
This is great for negative habits. Let's say drinking alcohol. At the start if the day, you haven't drank alcohol so it should be Done by default. Then change it to Fail if you do drink.

**Comment by herrbrixter** [comment]
Related to #696
